### PR TITLE
RAT-362, RAT-345: Fix gitignore directory handling - switch to relative paths in underlying library

### PR DESCRIPTION
**Pull request by nielsbasjes**
This fixes https://issues.apache.org/jira/browse/RAT-362

### CODE REVIEW

**Review by Claudenw — CHANGES_REQUESTED** [review]
@nielsbasjes do you have a test that shows failure before this fix is applied?

Please include such a test in your pull request.

**Comment by nielsbasjes** [comment]
> @nielsbasjes do you have a test that shows failure before this fix is applied?

Yes.
In this case it is a problem in the upstream library (which I wrote) where I have included extensive tests to prove it now works correctly: https://github.com/nielsbasjes/codeowners/blob/main/gitignore-reader/src/test/java/nl/basjes/gitignore/TestBugreports.java#L32-L68

> Please include such a test in your pull request.
Done.

Note that on my machine none of the tests in RatCheckMojoTest run automatically during the build. I can run them manually though. Since this is using a very old Junit API, I'm unsure on how to fix this (separate) issue.

**Comment by nielsbasjes** [comment]
For the tests I created this https://issues.apache.org/jira/browse/RAT-367

**Comment by nielsbasjes** [comment]
@Claudenw I think the testing issue https://github.com/apache/creadur-rat/pull/230 should be fixed first because otherwise there is no way to show this actually works on all JDK versions on all platforms (i.e. also Windows).

**Comment by ottlinger** [comment]
@nielsbasjes pls rebase your branch in order for all build checks to run again (as RAT-367 is merged now). Thanks

File: apache-rat-plugin/src/test/resources/unit/RAT-362-GitIgnore/pom.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>org.apache.rat.test</groupId>
+  <artifactId>RAT362</artifactId>
```
Comment (ottlinger) [inline apache-rat-plugin/src/test/resources/unit/RAT-362-GitIgnore/pom.xml:21]: Please change to RAT-362 for better consistency with directory and resource names. Thx.

**Comment by nielsbasjes** [comment]
I suspect the new test will fail on Windows because I expect to get a drive letter in the middle.

File: apache-rat-plugin/src/test/resources/unit/RAT-362-GitIgnore/pom.xml (hunk 1 above)
Comment (nielsbasjes) [reply apache-rat-plugin/src/test/resources/unit/RAT-362-GitIgnore/pom.xml]: Fixed

**Review by ottlinger — APPROVED** [review]
Thanks, I'll add a changelog entry and merge your changes.

**Comment by ottlinger** [comment]
@Claudenw I do hope it is fine with you.

**Pull request merged**